src/components/WebPlayBack.tsx

```diff
@@ -15,32 +21,68 @@ function WebPlayback(props: any) {
         //'ts-ignore' is a Bandaid fix see issue on GitHub to create a 'Type Declaration File'.
         // @ts-ignore
         window.onSpotifyWebPlaybackSDKReady = () => {
-            const token = 'My Access Token';
-            //@ts-ignore
+            const token = 'BQCQQz1SLEaC7WIFgytE1srWTIIZSqZXgRPqWYE_CjH3Tgf4UAk-ennMkN8zAKaeQ22GLljT3F_T_EUjHUD2hqzIS8O8q92AwWIDsNa-5rMdyF3-MqcRnaJr377QW8E9k4GusUcFQk6glcDR4Kx9mq5aTPF1UBKM5QM';
             //For instantiating Spotify Player object.
+            //@ts-ignore
             const player = new window.Spotify.Player({
                 name: 'WebPlayBack SDK',
                 getOAuthToken: (cb: (arg:string) => any) => { cb(token); },
                 volume: 0.5
             });
-
-            player.addListener('ready', ({device_id}: {device_id: number}) => {
+            
+            player.addListener('ready', ({device_id}: {device_id: string}) => {
                 console.log('Ready with Device ID', (device_id));
+                //@ts-ignore
+                player.getCurrentState().then(state => {
+                    if (!state) {
+                        console.error('User is not playing music through the Web Playback SDK');
+                        return;
+                    }
+                    const updatedPlayerInfo: IPlayerUIProps = {
+                        trackName : state.track_window.name,
+                        albumName : state.track_window.album.name,
+                        artistName : state.track_window.artists[0].name
+                    }
+                    updatePlayerInfo(updatedPlayerInfo);
+                });
             });
 
-            player.addListener('not_ready', ({device_id}: {device_id: number}) => {
+            player.addListener('not_ready', ({device_id}: {device_id: string}) => {
                 console.log('Device ID has gone offline', device_id);
             });
 
+            //@ts-ignore
+            player.addListener('player_state_changed', (state) => {
+                console.log(state);
+                const {track_window: { current_track }} = state;
+
+                //Creating playerInfo object for props
+                const updatedPlayerInfo: IPlayerUIProps = {
```

Comment:
We have some repeated logic here and on line 36 where we transform a state update to the player information that we need. Can we create one function for doing these and call it from `player.addListener('player_state_changed')` and `player.getCurrentState().then`?

Reply:
Great! Thanks. I have now consolidated the code. Look at line 52 for a new update PlayerInfo method.

Reply:
Thanks for making the change! Looks good :smile: